ecmd-core/capi/ecmdDataBufferBase.C

```diff
@@ -1109,6 +1117,62 @@ uint32_t ecmdDataBufferBase::reverse() {
   return rc;
 }
 
+
+// This is really the same code that reverse() uses above, except it's calling fast_reverse_bytes() instead of fast_reverse32()
+// It also has an added check for byte alignment.
+uint32_t ecmdDataBufferBase::byteReverse() {
+  uint32_t rc = ECMD_DBUF_SUCCESS;
+  uint32_t l_words=0;
+  uint32_t l_slop =  (iv_NumBits % UNIT_SZ);
+
+  if (iv_NumBits % 8)
+  {
+      ETRAC("Buffer must be byte aligned to use byteReverse()");
+      RETURN_ERROR(ECMD_DBUF_INVALID_ARGS);
+  }
+
+  if (l_slop)
+    l_words = iv_NumBits/32+1;
+  else
+    l_words = iv_NumBits/32;
+
+  // reverse words
+  for (uint32_t i=0;i< l_words/2;i++){
+    uint32_t l_tmp = fast_reverse_bytes(iv_Data[l_words-1-i]); 
+    iv_Data[l_words-1-i] = fast_reverse_bytes(iv_Data[i]);
+    iv_Data[i] = l_tmp;
+  }
+
+  // if odd number of words, reverse middle word; if only 1 word, reverse this word
+  if (l_words&1){
+    iv_Data[l_words/2 ] = fast_reverse_bytes(iv_Data[l_words/2]);
+  }
+
+  // now account for slop
+  if (l_slop != 0){
+
```

Comment:
At this point all the data is in the right order.. you might just have some leading 0's to get rid of if not word aligned. ie.. 0000FEEDB0B0DEADBEEF

Could you cast the buffer to uint8_t.. advance the pointer by the number of slop bytes.. then cast it back to uint32_t?  Would that put the start of your buffer at the start of the data and save all the masking/shifting the code is doing now?

Reply:
There might be optimizations that are possible, but this was just copied from the logic in the reverse() method which looks like it hasn't been touched in nearly 15 years.  At this point, I'd rather not disrupt a stable code path.